src/passport-saml/saml.ts

```diff
@@ -716,18 +742,18 @@ class SAML {
 
         const xmlencOptions = { key: this.options.decryptionPvk };
         return util.promisify(xmlenc.decrypt).bind(xmlenc)(encryptedAssertionXml, xmlencOptions)
-        .then(decryptedXml => {
+        .then((decryptedXml: string) => {
           const decryptedDoc = new xmldom.DOMParser().parseFromString(decryptedXml);
-          const decryptedAssertions = xmlCrypto.xpath(decryptedDoc, "/*[local-name()='Assertion']");
+          const decryptedAssertions = xmlCrypto.xpath(decryptedDoc, "/*[local-name()='Assertion']") as HTMLElement[];
```

Comment:
Again, it seems we might be doing something sub-optimal or the types for `xmlCrypto` need to be updated.

Reply:
no, xmlCrypto types are correct, they don't have any insights about input value (does it select an attribute or an element) and we adding restrictions is our responsibility